firestore/integration_test_internal/src/aggregate_query_snapshot_test.cc

```diff
@@ -46,90 +46,339 @@ std::size_t AggregateQuerySnapshotHash(const AggregateQuerySnapshot& snapshot) {
+  const AggregateQuerySnapshot snapshot = AggregateQuerySnapshot();
+
+  EXPECT_EQ(snapshot.count(), 0);
+  EXPECT_EQ(snapshot.query(), AggregateQuery());
+  EXPECT_FALSE(snapshot.is_valid());
+
+  AggregateQuerySnapshot snapshot_copy_dest;
+
+  EXPECT_EQ(snapshot_copy_dest.count(), 0);
+  EXPECT_EQ(snapshot_copy_dest.query(), AggregateQuery());
+  EXPECT_FALSE(snapshot_copy_dest.is_valid());
+
+  snapshot_copy_dest = AggregateQuerySnapshot();
+
+  EXPECT_EQ(snapshot_copy_dest.count(), 0);
+  EXPECT_EQ(snapshot_copy_dest.query(), AggregateQuery());
+  EXPECT_FALSE(snapshot_copy_dest.is_valid());
+
+  EXPECT_EQ(snapshot.count(), 0);
+  EXPECT_EQ(snapshot.query(), AggregateQuery());
+  EXPECT_FALSE(snapshot.is_valid());
+}
+
+TEST_F(
+    AggregateQuerySnapshotTest,
+    ValidObjectCopyAssignmentOperatorAppliedToValidObjectReturnsEqualObject) {
+  const AggregateQuerySnapshot snapshot;
+
+  EXPECT_EQ(snapshot.count(), 0);
+  EXPECT_EQ(snapshot.query(), AggregateQuery());
+  EXPECT_FALSE(snapshot.is_valid());
+
+  const AggregateQuery aggregate_query =
+      TestFirestore()->Collection("foo").Limit(10).Count();
+
+  AggregateQuerySnapshot snapshot_copy_dest =
+      TestAggregateQuerySnapshot(aggregate_query, 7);
+
+  EXPECT_EQ(snapshot_copy_dest.count(), 7);
+  EXPECT_EQ(snapshot_copy_dest.query(), aggregate_query);
+  EXPECT_TRUE(snapshot_copy_dest.is_valid());
+
+  snapshot_copy_dest = snapshot;
+
+  EXPECT_EQ(snapshot.count(), 0);
+  EXPECT_EQ(snapshot.query(), AggregateQuery());
+  EXPECT_FALSE(snapshot.is_valid());
+
+  EXPECT_EQ(snapshot_copy_dest.count(), 0);
+  EXPECT_EQ(snapshot_copy_dest.query(), AggregateQuery());
+  EXPECT_FALSE(snapshot_copy_dest.is_valid());
+}
+
+TEST_F(
+    AggregateQuerySnapshotTest,
+    ValidObjectCopyAssignmentOperatorAppliedToDefaultObjectReturnsEqualObject) {
+  const AggregateQuery aggregate_query1 =
+      TestFirestore()->Collection("foo").Limit(10).Count();
+  const AggregateQuery aggregate_query2 =
+      TestFirestore()->Collection("bar").Limit(20).Count();
+
+  const AggregateQuerySnapshot snapshot =
+      TestAggregateQuerySnapshot(aggregate_query1, 1);
+
+  EXPECT_EQ(snapshot.count(), 1);
+  EXPECT_EQ(snapshot.query(), aggregate_query1);
+  EXPECT_TRUE(snapshot.is_valid());
+
+  AggregateQuerySnapshot snapshot_copy_dest =
+      TestAggregateQuerySnapshot(aggregate_query2, 2);
+
+  EXPECT_EQ(snapshot_copy_dest.count(), 2);
+  EXPECT_EQ(snapshot_copy_dest.query(), aggregate_query2);
+  EXPECT_TRUE(snapshot_copy_dest.is_valid());
+
+  snapshot_copy_dest = snapshot;
+
+  EXPECT_EQ(snapshot.count(), 1);
+  EXPECT_EQ(snapshot.query(), aggregate_query1);
+  EXPECT_TRUE(snapshot.is_valid());
+
+  EXPECT_EQ(snapshot_copy_dest.count(), 1);
+  EXPECT_EQ(snapshot_copy_dest.query(), aggregate_query1);
+  EXPECT_TRUE(snapshot_copy_dest.is_valid());
+}
+
+TEST_F(AggregateQuerySnapshotTest,
+       CopyAssignmentAppliedSelfReturnsEqualObject) {
```

Comment:
Please add tests for move-to-self as well. The state of an object after being moved to itself is not defined: it could be either in the "moved-from" state or "moved-to" state. Since it's a bit of a degenerate situation, I'd just assume it's in the moved-from state, which means the only thing you need to test is that it doesn't crash when its destructor is run.